jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/DockerCredentialHelper.java

```diff
@@ -131,7 +153,9 @@ private Credential retrieve(List<String> credentialHelperCommand)
       throws IOException, CredentialHelperUnhandledServerUrlException,
           CredentialHelperNotFoundException {
     try {
-      Process process = processBuilderFactory.apply(credentialHelperCommand).start();
+      ProcessBuilder processBuilder = processBuilderFactory.apply(credentialHelperCommand);
+      processBuilder.environment().putAll(environment);
```

Comment:
One thing I'm curious about here is if a user sets an environment variable for the cred helper from the command line and also sets the same environment variable in their build file. The environment variable set in the build file will take precedence - is that okay? My initial thought is no...

Reply:
> environment variable for the cred helper from the command line

You're talking about the environment variables seen from the JVM process (i.e., what you get with `processBuilder.environment()`, which is same as `System.getenv()`), right? I think it is right that the variables configured through Jib parameters overwrite the environment inherited from the JVM.